source/PlayerInfoPanel.cpp

```diff
@@ -246,80 +316,89 @@ bool PlayerInfoPanel::KeyDown(SDL_Keycode key, Uint16 mod, const Command &comman
 			}
 			
 			// Clamp the destination index to the end of the ships list.
-			size_t moved = allSelected.size();
-			toIndex = min(player.Ships().size() - moved, toIndex);
-			selectedIndex = player.ReorderShips(allSelected, toIndex);
+			size_t moved = panelState.AllSelected().size();
+			toIndex = min(panelState.Ships().size() - moved, toIndex);
+			player.ReorderShips(panelState.Ships());
+			// The order has now changed, so the ships are no longer sorted.
+			panelState.SetCurrentSort(nullptr);
+			
+			int selectedIndex = player.ReorderShips(panelState.AllSelected(), toIndex);
```

Comment:
It's needed because if the player sorted the ships, but didn't save the order and later manually reorders them, they should be reordered from the order currently shown on screen (sorted) and not the last saved order.

Reply:
Yes, but we should write code that does not look like it is a bug. If that's as simple  as changing a method name, we should do that.

We don't technically need these two methods to be overloads (we always know which PlayerInfo method we want to call), so let's rename the whole-order one, e.g. `SaveShipOrder` or equivalent.